### DISCUSSION TITLE
randint implementation for BOOST Move hub

### DISCUSSION THREAD

**Discussion by dlech** [pybricks/support#319] [Ideas]
Due to limited flash memory, the BOOST Move hub doesn't have the `urandom` module enabled. Here is a quick hack to implement a decent [randint](https://docs.python.org/3/library/random.html#random.randint) function.

```python
_rand = hub.battery.voltage() + hub.battery.current()  # seed

# Return a random integer N such that a <= N <= b.
def randint(a, b):
    global _rand
    _rand = 75 * _rand % 65537  # Lehmer
    return _rand * (b - a + 1) // 65537 + a
```

**Comment by JorgePe** [comment]
THANK YOU!!!

**Comment by JobadSayas** [comment]
It always gets the same number...

**Reply by dlech** [reply]
It is working for me with firmware v3.2.3.

```python
from pybricks.hubs import MoveHub

hub = MoveHub()

_rand = hub.battery.voltage() + hub.battery.current()  # seed

# Return a random integer N such that a <= N <= b.
def randint(a, b):
    global _rand
    _rand = 75 * _rand % 65537  # Lehmer
    return _rand * (b - a + 1) // 65537 + a

print([randint(0, 10) for _ in range(10)])
```

prints

```
[8, 4, 8, 9, 8, 2, 1, 5, 3, 8]
```

**Reply by laurensvalk** [reply]
I had the same problem as @yobsh.

It often gives the exact same sequence because voltage + current produces often produce the same seed. The voltage is averaged now, so it is not as random as it used to be.

This made the example somewhat unhelpful so it was removed from the docs.

Maybe we could have `hub.system.uptime()` or something like that if we want a source that appears more random between script runs.

**Comment by bcfletcher** [comment]
the variant  i use is:

def randint(a, b):
    _rand = hub.battery.voltage() + hub.battery.current() + drive_base.distance() + int(sw.time())
    _rand = 75 * _rand % 65537 
    return _rand * (b - a + 1) // 65537 + a


this assumes you have a unit in motion and i  reset the StopWatch  at the beginning of the program:

sw = StopWatch()
print("Reset StopWatch")
sw.reset()
print("Resume StopWatch")
sw.resume()
